src/state/actions/jobActions.ts

```diff
@@ -1,12 +1,24 @@
-import { fetchJobsData } from "../../api/jobApi"
+import { fetchJobsData ,fetchJobsDataByPage} from "../../api/jobApi"
 import { Dispatch } from 'redux'
-import { getJobsAction, getJobsLoading, GET_JOBS, GET_JOBS_ERROR, GET_JOBS_LOADING } from "../constants/jobConstant";
+import { getJobsAction, getJobsLoading, GET_JOBS, GET_JOBS_ERROR, GET_JOBS_LOADING ,GET_PAGINATION_JOBS} from "../constants/jobConstant";
 
 export const getData = (name: string) => async (dispatch: Dispatch<getJobsAction | getJobsLoading>) => {
```

Comment:
Do we need to use this function anymore ? if not we can remove it.
also please check and remove unused codes at other places

Reply:
> At the initial page, we should call the api to get data of page 1, not slice the job data to get that, right ?

I fixed all of them according to your preview. please check it again. thanks